src/libraries/System.DirectoryServices.Protocols/tests/DirectoryServicesTestHelpers.cs

```diff
@@ -24,26 +27,43 @@ public static bool IsLibLdapInstalled
                 {
                     try
                     {
-                        // Attempt PInvoking into libldap
-                        IntPtr handle = ber_alloc(1);
-                        ber_free(handle, 1);
+                        // Attempt PInvoking into libldap on Linux
+                        IntPtr handle = ber_alloc_Linux(1);
+                        ber_free_Linux(handle, 1);
                         _isLibLdapInstalled = true;
                     }
                     catch (Exception)
                     {
-                        _isLibLdapInstalled = false;
+                        try
+                        {
+                            // Attempt PInvoking into libldap on OSX
+                            IntPtr handle = ber_alloc_OSX(1);
+                            ber_free_OSX(handle, 1);
+                            _isLibLdapInstalled = true;
+                        }
+                        catch (Exception)
+                        {
+                            _isLibLdapInstalled = false;
+                        }
                     }
                 }
                 return _isLibLdapInstalled.Value;
             }
         }
 
-        internal const string OpenLdap = "libldap-2.4.so.2";
+        internal const string OpenLdapLinux = "libldap-2.4.so.2";
+        internal const string OpenLdapOSX = "libldap";
 
-        [DllImport(OpenLdap, EntryPoint = "ber_alloc_t", CharSet = CharSet.Ansi)]
-        internal static extern IntPtr ber_alloc(int option);
+        [DllImport(OpenLdapLinux, EntryPoint = "ber_alloc_t", CharSet = CharSet.Ansi)]
+        internal static extern IntPtr ber_alloc_Linux(int option);
 
-        [DllImport(OpenLdap, EntryPoint = "ber_free", CharSet = CharSet.Ansi)]
-        public static extern IntPtr ber_free([In] IntPtr berelement, int option);
+        [DllImport(OpenLdapLinux, EntryPoint = "ber_free", CharSet = CharSet.Ansi)]
+        public static extern IntPtr ber_free_Linux([In] IntPtr berelement, int option);
+
+        [DllImport(OpenLdapOSX, EntryPoint = "ber_alloc_t", CharSet = CharSet.Ansi)]
```

Comment:
Hi! Maybe use  native library definition instead of ber method's duplicates. For example:
BerLibrary = "lber-2.4.so.2" for Linux
BerLibrary = "libldap" for Osx

Reply:
I believe this is same discussion here: https://github.com/dotnet/runtime/pull/36669#discussion_r426909921

Reply:
mm I think that is not exactly right, I believe @flamencist is suggesting to use lber library for ber operations instead of libldap as we do today (for both test code and product code). I think it is fine to fix that but that should go in a separate PR so I will send a follow up for that.

Reply:
I see now.  Does this code actually need to call these APIs or is it just trying to call them to find out which lib is present?  If the latter then @safern's suggestion seems like a better solution since it avoids the BER lib discrepancy entirely.

Reply:
It is the latter, so I will apply @safern's comments 😄